packages/bp/src/common/utils.ts

```diff
@@ -29,3 +29,25 @@ export const getErrorMessage = (error: Error | string | unknown): string => {
 
   return ''
 }
+
+const regex = {
+  illegalFile: /[\/\?<>\\:\*\|"]/,
+  illegalPath: /[\?<>:\*\|"]/,
+  control: /[\x00-\x1f\x80-\x9f]/,
+  reserved: /\.\.+?(\/|\\)/g
+}
+
+export const isValid = (input: string, type: 'file' | 'path') => {
+  if (regex.control.test(input) || input.match(regex.reserved)) {
```

Comment:
I know this function works because I've seen the tests, but I have a few concerns with this function's implementation and signature:

1. Is there a difference between `regex.test()` and `string.match()` ? Why don't you use 2 different functions on the same line?

2. Can't you merge `control` and `illegalFile` regexes? Both seem to make sure invalid characters are not used.

3. Why don't you test the `reserved` regex only when `type === path` ? Forward and backward slashes are not allowed in file names anyway.

4. Is there any place in the codebase where we call this function with `type = 'file'` ? If not can we change the function signature to something like `isValidPath(path: string)` ?

Reply:
1. Regex test is global so if you test the same string twice, you will get different results. The first 3 regex matches the whole path, while the second may match one or multiple times in the string
2. I hate playing with regex and this keeps the complexity low while being clear on what it does
3. Keeping the logic clearer this way, this is not used extensively so it would provide no advantage
4. Not yet, but I plan to change it in the future to use this method by the ghost. Similar code is already in the ghost, but I wanted to use it as a common file